packages/flutter/test/material/slider_theme_test.dart

```diff
@@ -161,7 +174,146 @@ void main() {
     expect(lerp.valueIndicatorTextStyle.color, equals(middleGrey.withAlpha(0xff)));
   });
 
-  testWidgets('Default slider thumb shape draws correctly', (WidgetTester tester) async {
+  testWidgets('Default slider track draws correctly', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData(
+      platform: TargetPlatform.android,
+      primarySwatch: Colors.blue,
+    );
+    final SliderThemeData sliderTheme = theme.sliderTheme.copyWith(thumbColor: Colors.red.shade500);
+    double value = 0.25;
+    Widget buildApp({
+      bool enabled = true,
+    }) {
+      final ValueChanged<double> onChanged = enabled ? (double d) => value = d : null;
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: MediaQuery(
+          data: MediaQueryData.fromWindow(window),
+          child: Material(
+            child: Center(
+              child: SliderTheme(
+                data: sliderTheme,
+                child: Slider(
+                  value: value,
+                  label: '$value',
+                  onChanged: onChanged,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildApp());
+
+    final RenderBox sliderBox = tester.firstRenderObject<RenderBox>(find.byType(Slider));
+
+    expect(
+        sliderBox,
+        paints
+          ..rect(rect: Rect.fromLTRB(16.0, 299.0, 208.0, 301.0), color: sliderTheme.activeTrackColor)
+          ..rect(rect: Rect.fromLTRB(208.0, 299.0, 784.0, 301.0), color: sliderTheme.inactiveTrackColor)
+    );
+
+    await tester.pumpWidget(buildApp(enabled: false));
+    await tester.pumpAndSettle(); // wait for disable animation
+    // The disabled thumb is smaller so the track has to paint longer to get
+    // to the edge.
+    expect(
+        sliderBox,
+        paints
+          ..rect(rect: Rect.fromLTRB(16.0, 299.0, 202.0, 301.0), color: sliderTheme.disabledActiveTrackColor)
+          ..rect(rect: Rect.fromLTRB(214.0, 299.0, 784.0, 301.0), color: sliderTheme.disabledInactiveTrackColor)
+    );
+  });
+
+  testWidgets('Default slider overlay draws correctly', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData(
+      platform: TargetPlatform.android,
+      primarySwatch: Colors.blue,
+    );
+    final SliderThemeData sliderTheme = theme.sliderTheme.copyWith(thumbColor: Colors.red.shade500);
+    double value = 0.25;
+    Widget buildApp({
+      bool enabled = true,
+    }) {
+      final ValueChanged<double> onChanged = enabled ? (double d) => value = d : null;
+      return Directionality(
```

Comment:
It's OK to must build a MaterialApp/Scaffold for tests like this

Reply:
I made to be consistent with other tests (using the "Directionality -> MediaQuery -> Material -> Center -> SLiderTheme -> Slider" tree structure.) Is this one different?

Reply:
No, just a suggestion since pasting things together by hand is tedious and doesn't actually test the common case (MaterialApp/Scaffold is the common case).